src/components/tiles/SwapDetailsTiles.tsx

```diff
@@ -89,7 +90,8 @@ export function SwapDetailsTiles(props: Props) {
   )
   if (destinationDenomination == null) return null
 
-  const sourceAmount = convertNativeToDisplay(walletDefaultDenom.multiplier)(spendTargets[0].nativeAmount)
+  const sourceNativeAmount = sub(abs(transaction.nativeAmount), transaction.networkFee)
```

Comment:
This math won't work for tokens. You'll need to check if the currencyCode matches the mainnet before subtracting the fee. Also noting that we need to add tokenId to EdgeTxSwap